android/src/main/java/org/mozilla/mozstumbler/client/ObservedLocationsReceiver.java

```diff
@@ -164,13 +159,13 @@ public void onReceive(Context context, Intent intent) {
 
         synchronized(mCollectionPoints) {
             while (mCollectionPoints.size() > MAX_SIZE_OF_POINT_LISTS) {
-                mCollectionPoints.removeFirst();
+                mCollectionPoints.remove(0);
```

Comment:
I didn't see a way to do LinkedList foo = Collections.synchronized... so I had to change those convenience functions to generic List equivalents.
Let me look again, I am sure there is some way to use the synchronized list as a LinkedList